### ISSUE TITLE
All validation losses going up (`seg_loss`, `dfl_loss`, `box_loss`) except for `class_loss`

### ISSUE THREAD

**Issue by massisenergy** [ultralytics/ultralytics#20360]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm not sure why the model is learning about classification (decreasing loss) but other losses are increasing?

![Image](https://github.com/user-attachments/assets/d76a48a2-3b80-4d3a-b45d-d0e42481094c)

```
$ awk 'NR<61' w16_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val_imgsz1280ep600/w16_yolo_trLe_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val_imgsz1280ep600.log
New https://pypi.org/project/ultralytics/8.3.111 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.99 🚀 Python-3.11.7 torch-2.6.0+cu124 CUDA:1 (NVIDIA A40, 45518MiB)
engine/trainer: task=segment, mode=train, model=/data1/icmr/sourav/workspace/projects/Yolo/pretrainedModelsUltralytics/yolo11x-seg.pt, data=/data1/icmr/sourav/workspace/Datasets/yolo_trLe_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val/data.yaml, epochs=600, time=None, patience=600, batch=10, imgsz=1280, save=True, save_period=100, cache=False, device=1, workers=8, project=w16_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val_imgsz1280ep600, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=w16_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val_imgsz1280ep600/train
Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments
  0                  -1  1      2784  ultralytics.nn.modules.conv.Conv             [3, 96, 3, 2]
  1                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]
  2                  -1  2    389760  ultralytics.nn.modules.block.C3k2            [192, 384, 2, True, 0.25]
  3                  -1  1   1327872  ultralytics.nn.modules.conv.Conv             [384, 384, 3, 2]
  4                  -1  2   1553664  ultralytics.nn.modules.block.C3k2            [384, 768, 2, True, 0.25]
  5                  -1  1   5309952  ultralytics.nn.modules.conv.Conv             [768, 768, 3, 2]
  6                  -1  2   5022720  ultralytics.nn.modules.block.C3k2            [768, 768, 2, True]
  7                  -1  1   5309952  ultralytics.nn.modules.conv.Conv             [768, 768, 3, 2]
  8                  -1  2   5022720  ultralytics.nn.modules.block.C3k2            [768, 768, 2, True]
  9                  -1  1   1476864  ultralytics.nn.modules.block.SPPF            [768, 768, 5]
 10                  -1  2   3264768  ultralytics.nn.modules.block.C2PSA           [768, 768, 2]
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 13                  -1  2   5612544  ultralytics.nn.modules.block.C3k2            [1536, 768, 2, True]
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 16                  -1  2   1700352  ultralytics.nn.modules.block.C3k2            [1536, 384, 2, True]
 17                  -1  1   1327872  ultralytics.nn.modules.conv.Conv             [384, 384, 3, 2]
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 19                  -1  2   5317632  ultralytics.nn.modules.block.C3k2            [1152, 768, 2, True]
 20                  -1  1   5309952  ultralytics.nn.modules.conv.Conv             [768, 768, 3, 2]
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 22                  -1  2   5612544  ultralytics.nn.modules.block.C3k2            [1536, 768, 2, True]
 23        [16, 19, 22]  1   8323187  ultralytics.nn.modules.head.Segment          [1, 32, 384, [384, 768, 768]]
YOLO11x-seg summary: 379 layers, 62,051,411 parameters, 62,051,395 gradients, 319.7 GFLOPs

Transferred 1071/1077 items from pretrained weights
TensorBoard: Start with 'tensorboard --logdir w16_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val_imgsz1280ep600/train', view at http://localhost:6006/
Freezing layer 'model.23.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks passed ✅
train: Scanning /data1/icmr/sourav/workspace/Datasets/yolo_trLe_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val/labels/train... 2678 images, 2517 backgrounds, 0 corrupt: 100%|██████████| 5177/5177 [00:04<00:00, 1122.47it/s]
train: New cache created: /data1/icmr/sourav/workspace/Datasets/yolo_trLe_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val/labels/train.cache
val: Scanning /data1/icmr/sourav/workspace/Datasets/yolo_trLe_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val/labels/val... 52 images, 57 backgrounds, 0 corrupt: 100%|██████████| 102/102 [00:00<00:00, 1580.54it/s]
val: New cache created: /data1/icmr/sourav/workspace/Datasets/yolo_trLe_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val/labels/val.cache
Plotting labels to w16_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val_imgsz1280ep600/train/labels.jpg...
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: SGD(lr=0.01, momentum=0.9) with parameter groups 176 weight(decay=0.0), 187 weight(decay=0.00046875), 186 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 1280 train, 1280 val
Using 8 dataloader workers
Logging results to w16_241class1_24bkg_251b1pL_500w10b2_934w11b3_431w12b4_dummy2500train50val_imgsz1280ep600/train
Starting training for 600 epochs...

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
      1/600      40.5G      1.807      3.058      5.631      2.201          7       1280: 100%|██████████| 518/518 [12:22<00:00,  1.43s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:04<00:00,  1.47it/s]
                   all        102         46      0.686      0.523      0.633      0.287      0.689      0.531      0.629      0.296

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
      2/600        43G       1.74      2.538      2.047      2.004         10       1280: 100%|██████████| 518/518 [12:43<00:00,  1.47s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.61it/s]
                   all        102         46        0.4      0.522      0.445      0.157       0.42      0.543      0.456      0.163

```

### Additional

Method:
- transfer learning with yolo11x-seg.pt
- one class classification+segmentation
- with variable number of dummy images (without any corresponding .txt files)
- without the dummy images the training was going fine (all loss decreasing, also predictions were good)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @massisenergy, thank you for reaching out and providing detailed logs and background info! 🚀 This is an automated response to help streamline your issue—an Ultralytics engineer will assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) where common questions and troubleshooting strategies are outlined. If you haven’t already, please review the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for guidance on custom training setups.

Since this appears to be a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) if possible. This helps us investigate and resolve your issue more efficiently. Key details to include:
- A snippet or script that reproduces the increasing loss issue
- A few sample images (and label files) from your dataset
- Any relevant config files or hyperparameter changes

Join our community for faster feedback and collaborative support:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for sharing and learning with others

## Upgrade

Make sure you are running the latest `ultralytics` package and dependencies ([requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in the following verified environments (with all dependencies preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for your report! An Ultralytics engineer will review your issue and follow up soon.

**Comment by glenn-jocher** [comment]
This appears to be related to your use of dummy images without corresponding annotation files. While your model is improving at classification (decreasing cls_loss), it's struggling with localization and segmentation, leading to increasing box_loss, seg_loss, and dfl_loss.

The dummy images without annotations are likely confusing the model - it sees objects but has no ground truth for their locations or segments. During training, this creates a conflict where the model tries to predict boxes/masks for objects in unannotated images, but then gets penalized since there are no ground truth annotations to match with.

I'd recommend either:
1. Remove the dummy images from your dataset
2. Create proper annotations for all dummy images
3. If you need to keep unannotated images, consider using a semi-supervised approach with pseudo-labeling

For segmentation tasks especially, having accurate mask annotations is critical for proper training.

**Comment by glenn-jocher** [comment]
You're welcome! Let me address your questions:

1. Including these 2500 "no-detection" images is a good approach conceptually, but they need proper handling in the training pipeline.

2. Yes, exactly! Create empty `.txt` files for each dummy image. This explicitly tells the model "there are no objects in this image" rather than having the model confused by missing annotation files.

3. Semi-supervised learning would involve first training on your fully annotated data, then using that model to generate pseudo-labels for unlabeled images, filtering these pseudo-labels based on confidence, and finally training on both real and high-confidence pseudo-labels. For YOLO, you can find documentation on this approach in the [Ultralytics semi-supervised learning guide](https://github.com/ultralytics/ultralytics/issues/10248).

I'd strongly recommend trying the empty annotation files approach first, as it's straightforward and directly addresses your use case of true negative examples.

**Comment by massisenergy** [comment]
Thanks for confirming - I am actually trying out the empty annotation files approach. I think I explain with bit more detail to make sure that we're on the same page. 

**Available data**: 
 - more than 300000 images with ~5000 class 1. All are unannotated (no polygon mask). Rest can be considered dummy. No boxes/masks to be predicted on these ~25000 images (also no detection) by Yolo.
 - ~250 images with polygon annotation (are all class 1) available from experts.

**Goal**: For these 30000 images Yolo should 
 - detect the classes correctly
 - for class 1, predict polygon masks

**Approach**: 
1. Generate polygon masks for the class 1: 
   - Started with 250 images (~200 train, ~50 validation) to transfer learn with pretrained yolo11x-seg. Predict on the ~30000 images.
   - Took the images with correctly predicted and satisfactory polygon mask, included those in the next transfer learning training (started with pristine yolo11x-seg again).
   - Kept doing this cycle until mostly all the ~5000 images have polygon masks. The training set increase with each iteration and validation set remains small as in the first. The prediction set also decreases.
2. Improve the specificity (reduce false positive)
   - Used the last transfer-learnt model for prediction on the ~25000 images.
   - Took those which were falsely classified & annotated. Included in the training as dummy (with empty label.txt).


I am not sure how does it differ from the approach mention here [Ultralytics semi-supervised learning guide](https://github.com/ultralytics/ultralytics/issues/10248). For sure, I've not felt the need to implement new metrics calculation.

Attaching here some more training images at different iterations. 

**~200 images + ~250 pseudolabels**:
![](https://github.com/user-attachments/assets/f2d189d0-8aa7-427b-9b71-20d2eaccff29)

**4th Iteration**:

![Image](https://github.com/user-attachments/assets/d5e6e519-76b2-467b-9072-55abeb031d1c)

**7th Iteration**: Shared at the first message in this thread.

The recurrent theme is after certain epoch, all other losses do not improve (stay same or keep increasing), unlike the class loss. 

- Is there any problem (and potential improvements) in the training or this can be considered as a regular observation with approaches like mine.
- What could be the reasons?

**Comment by glenn-jocher** [comment]
Looking at your iterative pseudo-labeling approach and the recurring pattern in your losses, I believe what you're experiencing is common in semi-supervised segmentation learning. Your approach is similar to the guide I linked, though less formalized.

The increasing segmentation and localization losses compared to decreasing classification loss suggests:

1. **Error accumulation in pseudo-labels** - Each iteration potentially propagates and amplifies inaccuracies in boundary/mask definitions, while classification decisions remain relatively robust.

2. **Data imbalance effects** - With 5,000 positive vs. 25,000 negative examples, the model prioritizes classification correctness over boundary precision.

To address this pattern:

- Try adjusting loss weights to prioritize segmentation: increase `box=7.5` to something like `box=10.0` and perhaps also increase `seg_loss` weight
- Implement stricter confidence thresholds when selecting pseudo-labels for the next iteration
- Consider occasionally retraining from the original model using only your highest-quality pseudo-labels rather than continually building on previous iterations
- Reserve a small set of expert-annotated data for true validation that never enters your training pipeline

This pattern isn't necessarily a problem, but these adjustments might help balance the emphasis between classification and segmentation performance.
